lib/saluki-context/src/origin.rs

```diff
@@ -67,7 +70,7 @@ impl fmt::Display for OriginTagCardinality {
     }
 }
 
-/// Information about the entity where a metric originated from.
+/// A borrowed representation of an origin.
```

Comment:
Is this really a borrowed representation of an origin? It feels more like "potentially incomplete collection of borrowed things that are associated with an origin and can be used to look one up", which obviously won't work as a name. But when I see `Ref` I think of something more akin to a pointer/reference, i.e. that's very direct and specific. Anyway, this might be too nitpicky but figured it was worth mentioning just for the sake of making names as clear as possible.

Reply:
Hmm, no, that's a very fair point.

Let me noodle on other naming schemes.

Reply:
I adjusted `OriginRef<'a>` and `ExternalDataRef<'a>` to be `RawOrigin<'a>` and `ExternalDataRef<'a>`, respectively. While `ExternalDataRef<'a>` _is_ much closer to a pointer/reference to `ExternalData`, you can't actually get one from `ExternalData`, so it's never used as reference, only a zero-copy lookup key, same as `RawOrigin<'a>`.

Let me know what you think.